pkg/apis/autoscaling/v1/conversion_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"testing"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+func TestConvert_autoscaling_HorizontalPodAutoscalerSpec_To_v1_HorizontalPodAutoscalerSpec(t *testing.T) {
+
+	type args struct {
+		in  *autoscaling.HorizontalPodAutoscalerSpec
+		out *autoscalingv1.HorizontalPodAutoscalerSpec
+		s   conversion.Scope
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			// Test case for #101370
+			"TestConversionWithCPUAverageValueAndUtilizationBoth",
```

Comment:
Can you add two more test cases - one where each one of these is set individually, and then one where these different?  I would also expect to see out check the Metrics field (not just whether target cpu utilization is nil).

Reply:
Cases added. 
In my test,  all the cases passed after this change. And the case "TestConversionWithCPUAverageValueAndUtilizationBoth" failed with the old code.
PTAL, Thanks.